examples/tensorflow-profiling-examples/example_test.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import testing_utils
+from tensorflow.python.keras import initializers
+
+import example
+
+
+class LinearBlockFullTest(tf.test.TestCase):
+    def test_basic(self):
+       testing_utils.layer_test(example.LinearBlockFull, input_shape=(4, 32))
+
+    def test_output(self):
+        batch_size, dim, output_dim = (3, 4, 2)
+        testing_utils.layer_test(
+            example.LinearBlockFull,
+            kwargs={'units': output_dim},
+            input_data=np.ones((batch_size, dim)),
+            expected_output_dtype='float32'
+        )
+
+
+class LinearBlockTest(tf.test.TestCase):
+    def test_shape_default(self):
+        x = np.ones((4, 32))
+        layer = example.LinearBlock()
+        output = layer(x)
+        self.assertAllEqual(output.shape, (4, 32))
+
+    def test_shape(self):
+        batch_size, dim = (4, 15)
+        x = np.ones((batch_size, dim))
```

Comment:
nit: I think having `dim` be both the input dimension and output dimension makes this test harder to read. I suggest separate values so the reader knows the input dim has nothing to do with this test. Same thoughts apply to other tests.

Reply:
fixed